src/sentry/lang/javascript/processor.py

```diff
@@ -732,146 +653,163 @@ def expand_frames(self, frames, release):
         all_errors = []
         sourcemap_applied = False
 
-        for frame in frames:
-            errors = cache.get_errors(frame.abs_path)
-            if errors:
-                all_errors.extend(errors)
+        errors = cache.get_errors(frame['abs_path'])
+        if errors:
+            all_errors.extend(errors)
 
-            # can't fetch source if there's no filename present
-            if not frame.abs_path:
-                continue
+        # can't fetch source if there's no filename present
+        if not frame.get('abs_path'):
+            return
 
-            source = self.get_source(frame.abs_path, release)
-            if source is None:
-                logger.debug('No source found for %s', frame.abs_path)
-                continue
+        # This might fail but that's okay, we try with a different path a
+        # bit later down the road.
+        source = self.get_source(frame['abs_path'])
 
-            sourcemap_url, sourcemap_view = sourcemaps.get_link(frame.abs_path)
-            if sourcemap_view and frame.colno is None:
-                all_errors.append({
-                    'type': EventError.JS_NO_COLUMN,
-                    'url': expose_url(frame.abs_path),
-                })
-            elif sourcemap_view:
-                last_token = token
+        in_app = None
+        new_frame = dict(frame)
+        raw_frame = dict(frame)
 
-                if is_data_uri(sourcemap_url):
-                    sourcemap_label = frame.abs_path
-                else:
-                    sourcemap_label = sourcemap_url
+        sourcemap_url, sourcemap_view = sourcemaps.get_link(frame['abs_path'])
+        if sourcemap_view and frame.get('colno') is None:
```

Comment:
We should really follow up and more-heavily-comment some of this, e.g.

```
# If source map exists, but original source frame lacks column information,
# it's not possible to locate the original untransformed code - emit an error.
```

.. because reading it does not feel obvious to me at a glance.